### PR TITLE
feat: add data zoom to metric visualization

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #12658 

### Description:

- creates visualization/ folder
- moves types to that folder, along with the component
- adds new hook useChartZoom which includes a bunch of handlers to be used in recharts charts

Demo

https://github.com/user-attachments/assets/b5cf908a-038c-4ea9-ac12-fd915ea10109

> [!NOTE]
> Git is not smart enough to detect a file that was moved when it contains significant changes. 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+            resetZoom,
+        },
+        activeData,
+    } = useChartZoom({
+        data: timeSeriesData || [],
+    });
+
+    useEffect(() => {
+        resetZoom();
+    }, [data.comparisonRows, resetZoom]);
```
Comment (IrakliJani) [inline packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx:120]: why only comparison rows and not both?

**Review by IrakliJani — COMMENTED** [review]
approved code changes with one question

File: packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+            handleMouseUp,
+            resetZoom,
+        },
+        activeData,
+    } = useChartZoom({
+        data: timeSeriesData || [],
+    });
+
+    useEffect(() => {
+        resetZoom();
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx:112]: now, every time the data changes, the zoom will reset @IrakliJani . Thank you for pointing it out. I feel like this is ok bc `data` is a stable reference, if I understand this correctly

Comment (IrakliJani) [reply packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx:112]: makes sense!

**Pull request merged**